addons/mail/static/tests/mail_test_helpers_contains.js

```diff
@@ -617,10 +618,12 @@ class Contains {
         this.onFocus = () => this.runOnce("after focus");
         this.onScroll = () => this.runOnce("after scroll");
         if (!this.runOnce("immediately")) {
-            this.timer = setTimeout(
-                () => this.runOnce("Timeout of 3 seconds", { crashOnFail: true }),
-                3000
-            );
+            this.timer = setTimeout(() => {
+                this.timeoutCount++;
+                this.runOnce(`Timeout of ${this.timeoutCount * 0.25} seconds`, {
+                    crashOnFail: this.timeoutCount > 29,
+                });
+            }, 100);
```

Comment:
contains is deprecated but ok as I quick fix in the meantime.
The timer should be re-started at each step until the end (or use setInternal and clean it at the end).